impl/forms/CompressedForms.h

```diff
@@ -0,0 +1,305 @@
+#pragma once
+
+#include "mavis/Form.h"
+#include <map>
+#include <cinttypes>
+
+namespace mavis {
+
+/**
+ * C0-Form: INTENDED FOR DECODING USE -- probably not useful for general extraction
+ */
+class Form_C0
+{
+public:
+    enum idType : uint32_t
+    {
+        FUNC3 = 0,
+        IMM3,
+        RS1,
+        FUNC2A,
+        IMM2,
+        RD,
+        OPCODE,
+        __N
+    };
+
+    static const char * name;
+    static const FieldsType fields;
+    static const std::map<std::string, const Field &> fmap;
+    static const std::map<std::string, idType> imap;
+    static const FieldsType opcode_fields;
+    static const ImmediateType immediate_type;
```

Comment:
If we removed the `static` qualifier we easily could move these variables into the base class. But I don't think a derived class can define a static variable since it would mean that there is more than one instance of the variable. At least that's my understanding of what `static` means.

Reply:
Question: Is the `static` qualifier even useful here? I think we only ever create one instance of the class.

Reply:
Yes that's right so we can leave `name` as static and move the other members to the base class. I'll look at doing that in my next PR.